### PR TITLE
Bookshelf view

### PR DESCRIPTION
**Pull request by flagoon**
Hey @binygal I finished another part of my app. It took me much more time than I expected, but I was trying some new things with styled-components and failed miserably. My next step would be creating views for Archive and Add book form. Then mostly logic.

### CODE REVIEW

File: src/Components/MainArea/Bookshelf/Bookshelf.tsx [hunk 1]
```diff
@@ -1,5 +1,19 @@
 import React from 'react';
+import { IBooks } from '../../../typings/IBooks';
+import { Bookcase } from './Bookcase/Bookcase';
 
-const Bookshelf = () => <div>Bookshelf</div>;
+interface IProps {
```
Comment (binygal) [inline src/Components/MainArea/Bookshelf/Bookshelf.tsx]: `IProps` sounds like a very generic name. Let's try to find a name that indicates the actual meaning of this interface. Maybe `IBookshelfProps`?

File: src/Components/MainArea/Bookshelf/Bookcase/Bookcase.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import styled from 'styled-components';
+import { IBooks } from '../../../../typings/IBooks';
+
+interface IProps {
```
Comment (binygal) [inline src/Components/MainArea/Bookshelf/Bookcase/Bookcase.tsx]: `IProps` sounds like a very generic name. Let's try to find a name that indicates the actual meaning of this interface. Maybe `IBookProps`?

File: src/Components/MainArea/Bookshelf/Bookshelf.tsx [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
+    books: IBooks[];
+}
+
+const Bookshelf = ({ books }: IProps) => {
```
Comment (binygal) [inline src/Components/MainArea/Bookshelf/Bookshelf.tsx]: Destructuring is awesome in functions, but I found it to be not scaleable in function components. When you have visual only component very soon you will want to customize it with sending DOM properties using spread operator like that - `<div {...props}>` therefore I always prefer to pass props to the function component and then split it to props and domProps like this - `const { books, ...domProps} = props`

File: src/Containers/MainArea/Bookshelf/BookshelfContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import Bookshelf from '../../../Components/MainArea/Bookshelf/Bookshelf';
+import { IBooks } from '../../../typings/IBooks';
+
+interface IState {
+    books: IBooks[];
+}
+
+export default class BookshelfContainer extends React.Component<{}, IState> {
```
Comment (binygal) [inline src/Containers/MainArea/Bookshelf/BookshelfContainer.tsx]: Good usage of Container design! 😃

File: src/Components/MainArea/Bookshelf/Bookcase/Bookcase.styled.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import styled from 'styled-components';
```
Comment (binygal) [inline src/Components/MainArea/Bookshelf/Bookcase/Bookcase.styled.js:1]: nit: The risk in those files is that deleting the component itself, won't delete this code. I believe that [tree shaking](https://medium.com/@netxm/what-is-tree-shaking-de7c6be5cadd) will fix the bundle size, but you still have dead code in your repo.

**Review by binygal — APPROVED** [review]
LGTM please note the `nit` comments. Those are intended for you to learn. No need for a change.

File: src/Components/MainArea/Bookshelf/Bookshelf.tsx (hunk 1 above)
Comment (flagoon) [reply src/Components/MainArea/Bookshelf/Bookshelf.tsx]: My reasoning for this was, I wasn't using this interface anywhere else, just for the component the interface was created for. That's why more generic IProps and IStates was sufficient. I did correct this.

File: src/Components/MainArea/Bookshelf/Bookshelf.tsx (hunk 3 above)
Comment (flagoon) [reply src/Components/MainArea/Bookshelf/Bookshelf.tsx]: You right on this one. It looks fancy but could make problems in the future.

File: src/Components/MainArea/Bookshelf/Bookcase/Bookcase.styled.js (hunk 5 above)
Comment (flagoon) [reply src/Components/MainArea/Bookshelf/Bookcase/Bookcase.styled.js:1]: TBH I don't understand it. I know what tree shaking is (I know the concept). Is there a better way for keeping styled component? Do you suggest keeping styled-component in the component itself? When I want to delete the component, I would delete the whole folder anyway (or component and files connected to it). Maybe I'm lacking the experience to understand it. Please explain.

**Pull request merged**